### PR TITLE
Accept return as a valid answer to 'Edit the file again?'

### PR DESCRIPTION
**Pull request by sentriz**
## Description

when you used `funced` and there was an error when you finished editing the function, it asks the question
> Edit the file again? [Y/n]  

the uppercase `Y` of course being the default answer.
in pretty much every piece of software I've used that does this `Y/n` / `y/N` stuff - return means the default option, the capital letter

this PR does that

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

### CODE REVIEW

File: share/functions/funced.fish [hunk 1]
```diff
@@ -105,9 +105,7 @@ function funced --description 'Edit function definition'
                 echo # add a line between the parse error and the prompt
                 set -l repeat
                 set -l prompt (_ 'Edit the file again\? [Y/n]')
-                while test -z "$repeat"
-                    read -p "echo $prompt\  " repeat
-                end
+                read -p "echo $prompt\  " repeat
```
Comment (krader1961) [inline share/functions/funced.fish]: This is a step in the right direction. The problem is that it accepts any random string, as well as the empty string, as being an answer in the affirmative if it isn't some variation of "no". That's dangerous. Obviously this was an issue  in the original version and not something you introduced. Nonetheless, since you're touching this would you mind also explicitly testing for one of the acceptable affirmative answers (which include the empty string)?

**Comment by sentriz** [comment]
how about now?

**Comment by krader1961** [comment]
Merged. Thx.
